python/lsst/cp/pipe/cpCombine.py

```diff
@@ -400,17 +400,24 @@ def applyScale(self, exposure, scale=None):
         ----------
         exposure : `lsst.afw.image.Exposure`
             Exposure to scale.
+        bbox : `lsst.geom.Box2I`
+            BBox matching the segment of the exposure passed in.
         scale : `float` or `list` [`float`], optional
             Constant scale to divide the exposure by.
         """
         if scale is not None:
             mi = exposure.getMaskedImage()
             if isinstance(scale, list):
+                # Create a realization of the per-amp scales as an
+                # image we can take a subset of.  This may be slightly
+                # slower than only populating the region we care
+                # about, but this avoids needing to do arbitrary
+                # numbers of offsets, etc.
+                scaleExp = afwImage.MaskedImageF(exposure.getDetector().getBBox())
                 for amp, ampScale in zip(exposure.getDetector(), scale):
-                    ampIm = mi[amp.getBBox()]
-                    ampIm /= ampScale
-            else:
-                mi /= scale
+                    scaleExp.image[amp.getBBox()] = ampScale
```

Comment:
So this creates a fake full exposure and fills it with the amp scale values?

And I guess the performance penalty is that it has to do this every time we call applyScale?  Will the scale_list be different for every input exposure, or is there some caching that could be done?  (And feel free to tell me that this would be premature optimization anyway...)

Reply:
That's correct.  I thought of trying to be clever, but that seems like I'm just making more places to get geometry wrong.
We do instantiate a new full-size image for each `applyScale` call, which means we make `(Nrows / NsubRows)` copies of the same image for each input.  This isn't optimal, but pulling it out of the loop will cause the same memory issues the partial reads fixed.  The scales are different for each input exposure, so I'm not sure there's any way to optimize there.